src/cata_tiles.h

```diff
@@ -25,9 +25,37 @@ class JsonObject;
 struct visibility_variables;
 
 /** Structures */
+struct sprite_variation {
+    int weight;
+    std::vector<int> sprite_ids;
+
+    sprite_variation()
+    {
+        weight = 1;
+        sprite_ids.clear();
+    }
+
+    sprite_variation(int id)
+    {
+        weight = 1;
+        sprite_ids.clear();
+        sprite_ids.push_back(id);
+    }
+
+    sprite_variation(int w, int id)
+    {
+        weight = w;
+        sprite_ids.clear();
```

Comment:
Why all those `clear` calls in the constructors? Do you think the vector is initialized to some content?

To make this clear: a vector is initialized by its default constructor to be empty. The `clear` calls here a re completely pointless.

And why not use the `weighted_list` class from src/weighted_list.h?